src/databaseAdapters/user.ts

```diff
@@ -33,21 +30,20 @@ class MongoUserAdapter implements DatabaseAdapter {
     await user.save();
   }
 
-  async setAvatarAndUpdateLibrary(username: string, avatarSet: S3AvatarSet) {
+  async approveAvatar(username: string, avatarSet: S3AvatarSet) {
     const user = await this.getUser(username);
-
-    user.avatarImgRes = avatarSet.resLink;
-    user.avatarImgSpy = avatarSet.spyLink;
-    user.avatarLibrary.push(avatarSet.avatarSetId);
-
     const librarySize = await getAvatarLibrarySizeForUser(
```

Comment:
On second thought, I don't think this guy should be here. It should be one level higher in the caller.